frontend/apps/docs/app/docs/sitemap.ts

```diff
@@ -0,0 +1,39 @@
+// biome-ignore lint/correctness/noNodejsModules: This file is used to generate the sitemap and is executed on the server-side only, requiring Node.js modules.
+import fs from 'node:fs'
+// biome-ignore lint/correctness/noNodejsModules: This file is used to generate the sitemap and is executed on the server-side only, requiring Node.js modules.
+import path from 'node:path'
+import type { MetadataRoute } from 'next'
+
+const getDocsPaths = (dir: string, basePath = '/docs'): string[] => {
+  const docsDir = path.join(process.cwd(), dir)
+  const files = fs.readdirSync(docsDir)
+
+  let paths: string[] = []
+
+  for (const file of files) {
+    const filePath = path.join(docsDir, file)
+    const stat = fs.statSync(filePath)
+
+    if (stat.isDirectory()) {
+      paths = paths.concat(
+        getDocsPaths(path.join(dir, file), path.join(basePath, file)),
+      )
+    } else if (file.endsWith('.mdx')) {
+      paths.push(`${basePath}/${file.replace(/\.mdx?$/, '')}`)
+    }
+  }
+
+  return paths
+}
```

Comment:
There is a `getPages()` API in fumadocs, so this should work.

https://fumadocs.vercel.app/docs/headless/source-api#get-pages

```diff
diff --git a/frontend/apps/docs/app/docs/sitemap.ts b/frontend/apps/docs/app/docs/sitemap.ts
index f2a44d51..0285fed7 100644
--- a/frontend/apps/docs/app/docs/sitemap.ts
+++ b/frontend/apps/docs/app/docs/sitemap.ts
@@ -2,6 +2,7 @@
 import fs from 'node:fs'
 // biome-ignore lint/correctness/noNodejsModules: This file is used to generate the sitemap and is executed on the server-side only, requiring Node.js modules.
 import path from 'node:path'
+import { source } from '@/lib/source'
 import type { MetadataRoute } from 'next'
 
 const getDocsPaths = (dir: string, basePath = '/docs'): string[] => {
@@ -27,6 +28,7 @@ const getDocsPaths = (dir: string, basePath = '/docs'): string[] => {
 }
 
 export default function sitemap(): MetadataRoute.Sitemap {
+  console.log({ pages: source.getPages() })
   const staticPaths = ['/docs']
   const docPaths = getDocsPaths('../docs/content/docs')
 
```

Server log ↓

```
@liam-hq/docs:dev:  GET /docs/sitemap.xml 200 in 349ms
@liam-hq/docs:dev:  ✓ Compiled in 320ms (2658 modules)
@liam-hq/docs:dev:  ✓ Compiled in 359ms (2658 modules)
@liam-hq/docs:dev: {
@liam-hq/docs:dev:   pages: [
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/debug',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs',
@liam-hq/docs:dev:       slugs: [],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/url-based-access',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build/drizzle',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build/getting-started',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build/postgresql',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build/prisma',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     },
@liam-hq/docs:dev:     {
@liam-hq/docs:dev:       file: [Object],
@liam-hq/docs:dev:       url: '/docs/static-build/rails',
@liam-hq/docs:dev:       slugs: [Array],
@liam-hq/docs:dev:       data: [Object],
@liam-hq/docs:dev:       locale: undefined
@liam-hq/docs:dev:     }
@liam-hq/docs:dev:   ]
@liam-hq/docs:dev: }
@liam-hq/docs:dev:  GET /docs/sitemap.xml 200 in 28ms
```

Reply:
Thanks to your advice, I could greatly reduce the code!♻️
https://github.com/liam-hq/liam/pull/392/commits/63a38c096f3dbb46da549344a89e486436c56fe5
https://github.com/liam-hq/liam/pull/392/commits/1720b61842d51701bc9c2ed5444551a7d7d9b0b3